MonoGame.Framework/GraphicsDeviceManager.cs

```diff
@@ -171,12 +171,13 @@ protected void OnDeviceDisposing(EventArgs e)
             Raise(DeviceDisposing, e);
         }
 
-        protected void OnDeviceResetting(EventArgs e)
+        // TODO these should be made protected, but currently WP calls these in SurfaceUpdateHandler
```

Comment:
Oh, my bad! I changed these to be protected to see where it was needed and the build didn't immediately fail (because I wasn't building WP), and when I did notice the fail I incorrectly changed them to public in stead of back to internal. Thanks! Did you have a chance to review the rest of this too? I think the easiest way is to focus on the tests I added and check if they are all decent since I basically did test-driven development. The workflow for this PR was to

1. Find issues (here on GitHub) related to GraphicsDeviceManager or come up with things that might be issues
2. write tests against XNA on this behavior and if they failed try something else and retest until they passed (so until I found the XNA behavior)
3. Check if they pass in MG and if not apply proper fixes until they did.

So checking the tests should cover most of it. I may have done some small things that were not based on a test here and there, but I can't really remember :p 

I think I can get around to fixing your remark and the conflicts tomorrow.